### ISSUE TITLE
[1.20.1] Entities in freeze_hurts_extra_types Entity Type Tags don't take extra damage when frozen

### ISSUE THREAD

**Issue by Insane96** [neoforged/NeoForge#701]
**Minecraft Version:** 1.20.1

**Steps to Reproduce:**
 1. Summon a Blaze in Powder snow
 2. He doesn't die in 4 freeze tick damage

**Description of issue:**
Seems like the patch made to `LivingEntity` breaks `freeze_hurts_extra_types` entity type tag, making them not take additional damage from freezing, due to the piece of code increasing the damage being enclosed in the `isDamageSourceBlocked` check.
This is `LivingEntity` from NeoForge 1.20.1 47.1.17
```java
if (pAmount > 0.0F && this.isDamageSourceBlocked(pSource)) {
            net.minecraftforge.event.entity.living.ShieldBlockEvent ev = net.minecraftforge.common.ForgeHooks.onShieldBlock(this, pSource, pAmount);
    if(!ev.isCanceled()) {
            if(ev.shieldTakesDamage()) this.hurtCurrentlyUsedShield(pAmount);
            f1 = ev.getBlockedDamage();
            pAmount -= ev.getBlockedDamage();
            if (!pSource.is(DamageTypeTags.IS_PROJECTILE)) {
               Entity entity = pSource.getDirectEntity();
               if (entity instanceof LivingEntity) {
                  LivingEntity livingentity = (LivingEntity)entity;
                  this.blockUsingShield(livingentity);
               }
            }

            flag = pAmount <= 0;
         }

         //This should not be here
         if (pSource.is(DamageTypeTags.IS_FREEZING) && this.getType().is(EntityTypeTags.FREEZE_HURTS_EXTRA_TYPES)) {
            pAmount *= 5.0F;
         }
    }
```

**Comment by Insane96** [comment]
From git blame seems like this has been fixed by PR #559, so this only affects 1.20.1

**Comment by Polarice3** [comment]
NeoForge 1.20.1 47.1.65 still has this problem. Not sure if still applicable on newer versions.
